File: Year-2023/LeetCode-January-Challenge/FindtheTownJudge.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    int findJudge(int n, vector<vector<int>>& trust) {
        vector<int> in(n + 1, 0), out(n + 1, 0);
        for(auto x: trust){
            in[x[1]]++;
            out[x[0]]++;
        }
        int res = -1;
        for(int i = 1; i <= n; ++i)
            if(in[i] == n - 1 && out[i] == 0) return res = i;
        return res;
    }
};
int main(){
    int n = 2; vector<vector<int>> trust = {{1,2}};
    Solution *s = new Solution();
    cout<<s->findJudge(n, trust)<<endl;
    return 0;
}

File: Year-2023/LeetCode-January-Challenge/FindtheTownJudge.java
class Solution {
    public int findJudge(int n, int[][] trust) {
        int[] in = new int[n + 1], out = new int[n + 1];
        for(int[] x: trust){
            in[x[1]]++;
            out[x[0]]++;
        }
        int res = -1;
        for(int i = 1; i <= n; ++i)
            if(in[i] == n - 1 && out[i] == 0) return res = i;
        return res;
    }
}
public class FindtheTownJudge {
    public static void main(String[] args) {
        int n = 2;
        int[][] trust = {{1,2}};
        Solution s = new Solution();
        System.out.println(s.findJudge(n, trust));
    }
}
